### PR TITLE
Add support for icu_locale and locale_provider postgresql_db module.

### LINKED ISSUE #666: Missing icu_locale and icu_provider parameters support in the DB module
##### SUMMARY

Since PostgreSQL 15, database creation parameters **icu_locale** and **icu_provider**  were added to PostgreSQL, but they are not supported by the DB module.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

- postgresql_db

##### ADDITIONAL INFORMATION

An attempt for a requested feature is in the  [pull request 667](https://github.com/ansible-collections/community.postgresql/pull/667) , hopefully it'll help.

### PR DESCRIPTION
**Pull request by aleszeleny**
##### SUMMARY

Add support for **icu_locale** and **locale_provider** to **postgresql_db module**

##### ISSUE TYPE

- Feature Pull Request - [issue 666](https://github.com/ansible-collections/community.postgresql/issues/666)

##### COMPONENT NAME

**postgresql_db**

##### ADDITIONAL INFORMATION

Hopefully, the pull request will be usable to show the missing feature.

closes #666

### CODE REVIEW

**Comment by aleszeleny** [comment]
> seems reasonable. just needs a changelog fragment please @aleszeleny

Hello. I'm sorry, would you like me to post the description here as a comment or add the "v3.3.1" description into `CHANGELOG.rst` ?
Thanks

File: CHANGELOG.rst
Comment (hunleyd) [inline CHANGELOG.rst:1]: Sorry, that's my bad. I meant to have you add a [changelog fragment](https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelogs.rst#examples). We should not edit CHANGELOG.rst directly, as we use antsibull-changelog to manage that file.

**Review by hunleyd — CHANGES_REQUESTED** [review]

**Comment by aleszeleny** [comment]
Hello,

that's fine, I'll remove it. Can I find somewhere a guide, whether it shall
be a new file (content format rules, a defined file name), or just post it
into GitHub conversation?

Thanks

Aleš Zelený

Aleš Zelený

Dne st 7. 2. 2024 16:35 uživatel Douglas J Hunley ***@***.***>
napsal:

File: plugins/modules/postgresql_db.py [hunk 1]
```diff
@@ -48,6 +48,19 @@
       - Must match LC_CTYPE of template database unless C(template0) is used as template.
     type: str
     default: ''
+  icu_locale:
+    description:
+      - Specifies the ICU locale (ICU_LOCALE) for the database default collation order and character classification, overriding the setting locale.
+      - The locale provider must be ICU. The default is the setting of locale if specified; otherwise the same setting as the template database.
+    type: str
+    default: ''
```
Comment (Andersson007) [inline plugins/modules/postgresql_db.py:56]:
```suggestion
    default: ''
    version_added: '3.4.0'
```

File: plugins/modules/postgresql_db.py [hunk 2]
```diff
@@ -48,6 +48,19 @@
...
+  locale_provider:
+    description:
+      - Specifies the provider to use for the default collation in this database (LOCALE_PROVIDER).
+      - Possible values are icu (if the server was built with ICU support) or libc.
+      - By default, the provider is the same as that of the template.
+    type: str
+    default: ''
```
Comment (Andersson007) [inline plugins/modules/postgresql_db.py:63]:
```suggestion
    default: ''
    version_added: '3.4.0'
```

**Review by Andersson007 — COMMENTED** [review]
@aleszeleny hello, thanks for the contribution! Couple of things from me

**Comment by aleszeleny** [comment]
> @aleszeleny thanks! also please revert the changes against the changelog and add a fragment as @hunleyd suggested https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelogs.rst#examples

Just tried to do this in commit d318950

File: changelogs/fragments/667-db_icu_provider.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+release_summary: |
+  This is the minor release of the ``community.postgresql`` collection.
+  This changelog contains all changes to the modules and plugins in this collection
+  that have been made after the previous release.
+
```
Comment (Andersson007) [inline changelogs/fragments/667-db_icu_provider.yml:5]:
```suggestion
```

**Review by hunleyd — APPROVED** [review]

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@aleszeleny thanks for the contribution! @hunleyd thanks for reviewing and approving!

**Comment by Andersson007** [comment]
@aleszeleny 3.4.0 has been released with your features, so you can update your local collection version using `ansible-galaxy` and enjoy it:)

Also you're welcome to join our forum group and the matrix channel, see the [communication guide](https://docs.ansible.com/ansible/devel/community/communication.html) or README for details.

**Pull request merged**